op-node/rollup/derive/engine_controller.go

```diff
@@ -177,7 +177,13 @@ func (e *EngineController) ConfirmPayload(ctx context.Context, agossip async.Asy
 		return nil, BlockInsertPrestateErr, fmt.Errorf("cannot complete payload building: not currently building a payload")
 	}
 	if e.buildingOnto.Hash != e.unsafeHead.Hash { // E.g. when safe-attributes consolidation fails, it will drop the existing work.
-		e.log.Warn("engine is building block that reorgs previous unsafe head", "onto", e.buildingOnto, "unsafe", e.unsafeHead)
+		// if we have a cached gossip payload, it means it was not yet imported, so we reuse it
+		// e.buildingOnto is cleared out when we reset the building state, which can happen even when a saved payload can be reused
+		if agossip.Get() != nil {
+			e.log.Warn("found reusable payload from async gossiper")
```

Comment:
are there any fields that we could add to the log here?

Reply:
Sure, I refactored this a bit so you can see the hash and the parent